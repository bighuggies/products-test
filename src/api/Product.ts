import { ICategory } from './Category';

export interface IProduct {
  id: string;
  title: string;
  description: string;
  categories: ICategory[];
}
