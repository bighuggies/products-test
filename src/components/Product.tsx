import * as React from 'react';

import { IProduct } from '../api/models/Product';

import { LinkButton } from './LinkButton';
import ProductDescription from './ProductDescription';

interface IProps {
  product: IProduct;
}

interface IState {
  isDescriptionShowing: boolean;
}

class Product extends React.PureComponent<IProps, IState> {
  public state = { isDescriptionShowing: false };

  public toggleDescription = () => {
    this.setState(prev => ({
      isDescriptionShowing: !prev.isDescriptionShowing
    }));
  };

  public render() {
    return (
      <div>
        <LinkButton
          className={this.state.isDescriptionShowing ? 'is-active' : ''}
          onClick={this.toggleDescription}
          aria-expanded={this.state.isDescriptionShowing}
          aria-controls={`${this.props.product.id}-description`}
        >
          {this.props.product.title}
        </LinkButton>

        {this.state.isDescriptionShowing && (
          <ProductDescription
            id={`${this.props.product.id}-description`}
            description={this.props.product.description}
          />
        )}
      </div>
    );
  }
}

export default Product;
