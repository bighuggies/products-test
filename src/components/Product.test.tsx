import { mount } from 'enzyme';
import * as React from 'react';

import Product from './Product';

test('renders without crashing', () => {
  const mounted = mount(
    <Product
      product={{ id: '1', title: 'p1', description: 'p1', categories: [] }}
    />
  );

  expect(mounted.debug()).toMatchSnapshot();
});

test('sets aria-expanded true when visible', () => {
  const mounted = mount(
    <Product
      product={{ id: '1', title: 'p1', description: 'p1', categories: [] }}
    />
  );

  mounted.setState({ isDescriptionShowing: true });

  expect(mounted.debug()).toMatchSnapshot();
});

test('sets aria-expanded false when not visible', () => {
  const mounted = mount(
    <Product
      product={{ id: '1', title: 'p1', description: 'p1', categories: [] }}
    />
  );

  mounted.setState({ isDescriptionShowing: false });

  expect(mounted.debug()).toMatchSnapshot();
});
