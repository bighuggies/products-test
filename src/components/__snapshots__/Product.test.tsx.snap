// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<Product product={{...}}>
  <div>
    <styled.button className=\\"\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
      <button className=\\"sc-bdVaJa fOgcfl\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
        p1
      </button>
    </styled.button>
  </div>
</Product>"
`;

exports[`sets aria-expanded false when not visible 1`] = `
"<Product product={{...}}>
  <div>
    <styled.button className=\\"\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
      <button className=\\"sc-bdVaJa fOgcfl\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
        p1
      </button>
    </styled.button>
  </div>
</Product>"
`;

exports[`sets aria-expanded true when visible 1`] = `
"<Product product={{...}}>
  <div>
    <styled.button className=\\"is-active\\" onClick={[Function]} aria-expanded={true} aria-controls=\\"1-description\\">
      <button className=\\"is-active sc-bdVaJa fOgcfl\\" onClick={[Function]} aria-expanded={true} aria-controls=\\"1-description\\">
        p1
      </button>
    </styled.button>
    <ProductDescription id=\\"1-description\\" description=\\"p1\\">
      <styled.p id=\\"1-description\\">
        <p className=\\"sc-bwzfXH Qoery\\" id=\\"1-description\\">
          p1
        </p>
      </styled.p>
    </ProductDescription>
  </div>
</Product>"
`;
